### ISSUE TITLE
Viewports stop updating

### ISSUE THREAD

**Issue by fabiansvara** [knossos-project/knossos#265]
This was triggered while zooming out. When this happens, i / o and mouse wheel zooming does not affect the XY viewport anymore and neither does moving in the dataset.

![knossos_zoom_bug_screenshot](https://cloud.githubusercontent.com/assets/7580864/8694768/78e04b2c-2ae1-11e5-8366-c617503db791.png)

**Comment by fabiansvara** [comment]
Just happened again, during zoom out (possibly exactly when crossing the mag boundary, not sure about this). Log since loading the last tracing:
[:0]    Debug:  QUrl( "file:///home/svara/work/mpi/Electron Microscopy/EM analysis/f0097 analysis/LVRD-Pre Analyis/re_tracet/lvr050_pre033/LVR050_Pre033.nml" )
[:0]    Debug: loading skeleton took:  50
[:0]    Debug: loading skeleton took:  17
[:0]    Critical: no slots 5 124 0
[:0]    Debug: loading skeleton took:  65
[:0]    Critical: no slots 5 124 0
[:0]    Debug: loading skeleton took:  5
[:0]    Critical: no slots 5 124 0
[:0]    Debug: selection hits:  5
[:0]    Debug: selection hits:  7
[:0]    Critical: no slots 5 124 0
[:0]    Warning: QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
[:0]    Warning: QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0
[:0]    Warning: QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0
[:0]    Warning: QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.
[:0]    Warning: QNetworkReplyImplPrivate::error: Internal problem, this method must only be called once.

[:0]    Critical: no slots 5 124 0

[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0
[:0]    Critical: no slots 5 124 0

**Comment by Optiligence** [comment]
The log is showing a problem, but missing cubes don’t explain why the viewport is unresponsive imo.

**Comment by fabiansvara** [comment]
This happens very regularly and is extremely annoying. Any suggestions for specific revisions that may have had an influence, so that I can start a seach for the commit that introduced this? (It started happening only recently...)

**Comment by orenshatz** [comment]
If it happens regularly use a screen capture so we know what has been done beforehand.
Never happened to me so far.
Possibly a hardware issue or OS specific?

**Comment by Optiligence** [comment]
I think i just experienced this. Maximizing and de-maximizing fixed it.

**Comment by orenshatz** [comment]
Possibly two different scenarios? Fabi's case looks like a memory corruptions, or else there wouldn't be so many critical/assertions. While your case seems more transient (or do you also see these log messages?)

**Comment by Optiligence** [comment]
the log has nothing to do with the issue

**Comment by orenshatz** [comment]
I assume the log is symptomatic to the deeper issue Fabi had, e.g. corruption, which might account for the other symptomatic update arrest.

**Comment by Optiligence** [comment]
That slots are missing is not a sign of memory corruption. the warnings from Qt are a known bug on their end.

**Comment by orenshatz** [comment]
AFAIK we should never miss a slot.

**Comment by Optiligence** [comment]
Right, but it just means that the managing logic is buggy (i assume so).

**Comment by orenshatz** [comment]
Or it means that the logic works on corrupt data.
Again, unless this assertion is quite common, even in the absence of other problems.
